pkg_resources/__init__.py

```diff
@@ -3047,12 +3047,12 @@ def reqs_for_extra(extra):
                 if not req.marker or req.marker.evaluate({'extra': extra}):
                     yield req
 
-        common = frozenset(reqs_for_extra(None))
+        common = dict.fromkeys(reqs_for_extra(None))
```

Comment:
There is a difference here: `frozenset` would yield an immutable data structure, while `dict` is mutable...

But I suppose it does not make a difference in this particular implementation because `common` is not being directly modified/assigned to another variable/data-structure until the end of the function, right? Or is there any chance other piece of code would accidentally mutate it?

Reply:
Yeah it's a bit less expressive a type but given that it's only used locally, I don't see that risk extending beyond the function. Were it to be 10s of lines of code, it might make sense to take precautions but since it's just a handful of statements, I think it's tolerable.

Reply:
Does it make sense to use something like `types.MappingProxy` to retain the original intention (of being read-only) and be future-proof?

Reply:
Done